src/jshint.js

```diff
@@ -2015,7 +2026,7 @@ var JSHINT = (function() {
               funct[v] = s["(global)"] ? "global" : "outer";
               break;
             case "const":
-              setprop(s, v, { unused: false });
+              s[v]["(type)"] = false;
```

Comment:
I've been trying to resolve this myself. While the current approach isn't really pretty, fixing it will require a more wide-spanning change that should be done independently (and preferably after this patch lands).

My plans are slightly different than your option "d", though. I think we should use a single label type, "binding" for all of these, then introduce two new attributes: "bindingType" (one of "var", "function", "let", or "const") and "used" (true or false). But if @caitp agrees with the order of operations, we can discuss that more in a follow-up issue.